CHANGELOG.md

```diff
@@ -1,122 +1,124 @@
-- Added the AllowUnresolvedParameters function flag that disables warnings for processing RPCs with unresolved parameters. This flag can be enabled through Blueprints or by adding a tag to the `UFUNCTION` macro.
-- A warning is shown if a cloud deployment is launched with the `manual_worker_connection_only` flag set to true
-- Server travel supported for single server game worlds. Does not currently support zoning or off-loading.
-- Enabled the SpatialOS toolbar for MacOS.
-- Improved workflow around schema generation issues and launching local builds. A warning will now show if attempting to run a local deployment after a schema error.
-- DeploymentLauncher can parse a .pb.json launch configuration.
-- DeploymentLauncher can launch a Simulated Player deployment independently from the target deployment.
+- A configurable Actor component 'SpatialPingComponent' is now available for PlayerControllers to measure round-trip ping to their current authoritative server-worker instance. The latest ping value can be accessed raw through the component via 'GetPing()' or otherwise via the rolling average stored in 'PlayerState'.
+- You can invoke the `GenerateSchema`, `GenerateSchemaAndSnapshots`, and `CookAndGenerateSchema` commandlets with the `-AdditionalSchemaCompilerArguments="..."` command line switch to output additional compiled schema formats. If you don't provide this switch, the output contains only the schema descriptor. This switch's value should be a subset of the arguments that you can pass to the schema compiler directly (for example `--bundle_out="path/to/bundle.sb"`). You can see a full list of possible values in the [schema compiler documentation](https://docs.improbable.io/reference/14.2/shared/schema/introduction#schema-compiler-cli-reference)
+- We've added the `AllowUnresolvedParameters` function flag that disables warnings for processing RPCs with unresolved parameters. To enable this flag, use Blueprints, or add a tag to the `UFUNCTION` macro.
+- There is now a warning if you launch a cloud deployment with the `manual_worker_connection_only` flag set to true.
+- We now support server travel for single server game worlds. Does not currently support zoning or off-loading.
+- We've improved the workflow relating to schema generation issues and launching local builds. There is now a warning if you try to run a local deployment after a schema error.
+- `DeploymentLauncher` can now launch a simulated player deployment independently from the target deployment.
 Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-deployment-name> <sim-deployment-name> <sim-deployment-json> <sim-deployment-region> <num-sim-players> <auto-connect>`
 - Added `HeartbeatTimeoutWithEditorSeconds` and use it if WITH_EDITOR is defined to prevent workers disconnecting when debugging while running in editor.
```

Comment:
What does “and use it” mean? The GDK uses it in the following circumstances? 

Also, “to prevent workers disconnecting when debugging while running in editor” - is this referring to PIE workers, or something else?

Reply:
1 - It's a variable name. Literally floating point value in a class
2 - In C++ you `defining` is actually a concept. You `define` a macro and then you can check if that macro is defined in later code with `if macro` e.g.:

```
#DEFINE WITH_EDITOR

#IF WITH_EDITOR
	do the thing...
#END IF WITH_EDITOR
```

Reply:
Thanks for your replies!

For 2, I still feel like my rewording makes more sense:

> `HeartbeatTimeoutWithEditorSeconds` is now true/now occurs/now does Z if Y is defined

instead of simply "added `HeartbeatTimeoutWithEditorSeconds`..."

What do you think?

Also, could you confirm the suggestion for the second sentence (see above)?

Reply:
This is actually a SpatialGDKSettings field (float) now.
```
	/**
	* Same as HeartbeatTimeoutSeconds, but used if WITH_EDITOR is defined.
	*/
	UPROPERTY(EditAnywhere, config, Category = "Heartbeat", meta = (DisplayName = "Heartbeat Timeout With Editor (seconds)"))
	float HeartbeatTimeoutWithEditorSeconds;
```
It means that there is a setting  `HeartbeatTimeoutWithEditorSeconds` which is only used if `WITH_EDITOR` is defined, which is only on Editor builds.